bin/git-forgit

```diff
@@ -19,10 +19,25 @@
 # shellcheck disable=2230
 SHELL="$(which bash)"
 export SHELL
-
 # Get absolute forgit path
 FORGIT=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)/$(basename -- "${BASH_SOURCE[0]}")
 
+# This function provides an abstraction for conditionally invoking
+# either `fzf` or `fzf-tmux` depending on the state of the current shell.
+#
+# While the fzf docs say that `fzf-tmux` is safe to use in scripts even outside
+# of a tmux session, we still employ defensive variable tests.
+do_fzf() {
+  if [ -n "${TMUX_PANE-}" ] && { [ "${FZF_TMUX:-0}" != 0 ] || [ -n "${FZF_TMUX_OPTS-}" ]; }; then
+   # We don't hardcode --exit-0 here because it causes flicker when using
+   # tmux + popups instead of panes. Could be revisited.
+   fzf-tmux "${FZF_TMUX_OPTS:--d${FZF_TMUX_HEIGHT:-40%}}" "$@"
+  else
+   fzf -0 "$@"
```

Comment:
I don't like hard-coding `-0` at this point for two reasons:

1. All other `fzf` arguments are defined somewhere else, this makes the final list of arguments harder to see in the code.
2. It actually changes behavior for some commands, since not every command had `-0` set in its argument list before.

The second point is questionable, though. I actually don't see any reason why we don't set `-0` on every command.
@wfxr @cjappl Do you have an idea? E.g. why do we set it on `gss` but not on `glo`?

Reply:
I'm gonna make a guess and say that the `-0` is only being set on commands where it's possible to have no results; `git log` is never gonna have no lines, for example.

That said it wouldn't *hurt* to have it everywhere. The "proper" fix is probably to introduce two separate `*DEFAULT_OPTS` variables for tmux and non-tmux but I was going for minimum-entropy with this changeset. I can do a more diligent refactor.

Reply:
https://github.com/wfxr/forgit/pull/304#discussion_r1186918216